### ISSUE TITLE
DI leading to error

### ISSUE THREAD

**Issue by intel352** [composer-unused/composer-unused#384]
## Describe the bug
Installed composer-unused, immediately ran into error. This is in a PHP 7.4 docker instance.

## Additional information
```
bash-5.1$ composer require composer-unused/composer-unused-plugin
Info from https://repo.packagist.org: #StandWithUkraine
Using version ^0.8.1 for composer-unused/composer-unused-plugin
./composer.json has been updated
Running composer update composer-unused/composer-unused-plugin
Loading composer repositories with package information
Updating dependencies
Lock file operations: 8 installs, 0 updates, 0 removals
  - Locking composer-unused/composer-unused-plugin (0.8.1)
  - Locking composer-unused/contracts (0.2.0)
  - Locking composer-unused/symbol-parser (0.1.9)
  - Locking icanhazstring/composer-unused (0.8.2)
  - Locking ondram/ci-detector (4.1.0)
  - Locking symfony/finder (v5.4.11)
  - Locking symfony/serializer (v5.4.11)
  - Locking webmozart/glob (4.6.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 8 installs, 0 updates, 0 removals
  - Installing webmozart/glob (4.6.0): Extracting archive
  - Installing symfony/serializer (v5.4.11): Extracting archive
  - Installing ondram/ci-detector (4.1.0): Extracting archive
  - Installing symfony/finder (v5.4.11): Extracting archive
  - Installing composer-unused/contracts (0.2.0): Extracting archive
  - Installing composer-unused/symbol-parser (0.1.9): Extracting archive
  - Installing icanhazstring/composer-unused (0.8.2): Extracting archive
composer-unused/composer-unused-plugin contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "composer-unused/composer-unused-plugin" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] y
  - Installing composer-unused/composer-unused-plugin (0.8.1): Extracting archive
    Install of composer-unused/composer-unused-plugin failed

In services.php line 30:
                                                                                                  
  [Error]                                                                                         
  Call to undefined function Symfony\Component\DependencyInjection\Loader\Configurator\service()  
                                                                                                  

Exception trace:
  at phar:///app/vendor/composer-unused/composer-unused-plugin/composer-unused.phar/config/services.php:30
 Symfony\Component\DependencyInjection\Loader\ProtectedPhpFileLoader::{closure}() at /app/vendor/symfony/dependency-injection/Loader/PhpFileLoader.php:50
 Symfony\Component\DependencyInjection\Loader\PhpFileLoader->load() at phar:///app/vendor/composer-unused/composer-unused-plugin/composer-unused.phar/config/container.php:12
 ComposerUnused\ComposerUnusedPlugin\UnusedPlugin::{closure}() at phar:///app/vendor/composer-unused/composer-unused-plugin/composer-unused.phar/config/container.php:21
 require() at /app/vendor/composer-unused/composer-unused-plugin/src/UnusedPlugin.php:36
 ComposerUnused\ComposerUnusedPlugin\UnusedPlugin->activate() at phar:///usr/local/bin/composer/src/Composer/Plugin/PluginManager.php:400
 Composer\Plugin\PluginManager->addPlugin() at phar:///usr/local/bin/composer/src/Composer/Plugin/PluginManager.php:291
 Composer\Plugin\PluginManager->registerPackage() at phar:///usr/local/bin/composer/src/Composer/Installer/PluginInstaller.php:85
 Composer\Installer\PluginInstaller->Composer\Installer\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/FulfilledPromise.php:28
 React\Promise\FulfilledPromise->then() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:134
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:168
 React\Promise\Promise->settle() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:231
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/FulfilledPromise.php:42
 React\Promise\FulfilledPromise->done() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:135
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:168
 React\Promise\Promise->settle() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:231
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/FulfilledPromise.php:42
 React\Promise\FulfilledPromise->done() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:66
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:168
 React\Promise\Promise->settle() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:231
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/FulfilledPromise.php:42
 React\Promise\FulfilledPromise->done() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:135
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:168
 React\Promise\Promise->settle() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:231
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/FulfilledPromise.php:42
 React\Promise\FulfilledPromise->done() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:135
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:168
 React\Promise\Promise->settle() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:231
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/FulfilledPromise.php:42
 React\Promise\FulfilledPromise->done() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:135
 React\Promise\Promise::React\Promise\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:168
 React\Promise\Promise->settle() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:231
 React\Promise\Promise::React\Promise\{closure}() at n/a:n/a
 call_user_func() at phar:///usr/local/bin/composer/src/Composer/Util/ProcessExecutor.php:321
 Composer\Util\ProcessExecutor->countActiveJobs() at phar:///usr/local/bin/composer/src/Composer/Util/Loop.php:98
 Composer\Util\Loop->wait() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:506
 Composer\Installer\InstallationManager->waitOnPromises() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:479
 Composer\Installer\InstallationManager->executeBatch() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:390
 Composer\Installer\InstallationManager->downloadAndExecuteBatch() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:282
 Composer\Installer\InstallationManager->execute() at phar:///usr/local/bin/composer/src/Composer/Installer.php:763
 Composer\Installer->doInstall() at phar:///usr/local/bin/composer/src/Composer/Installer.php:590
 Composer\Installer->doUpdate() at phar:///usr/local/bin/composer/src/Composer/Installer.php:279
 Composer\Installer->run() at phar:///usr/local/bin/composer/src/Composer/Command/RequireCommand.php:425
 Composer\Command\RequireCommand->doUpdate() at phar:///usr/local/bin/composer/src/Composer/Command/RequireCommand.php:288
 Composer\Command\RequireCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:1024
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:335
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:130
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:88
 require() at /usr/local/bin/composer:29

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>...]
```

**Comment by intel352** [comment]
@icanhazstring Looks like the symfony `service()` polyfill wasn't added until May 25, but the last release was Mar 22, so the `service()` fix isn't included.

**Comment by icanhazstring** [comment]
Thanks for the report and check. Let me publish a new release with it 😊

**Comment by icanhazstring** [comment]
Should be solved in `0.8.3`
